deepfence_worker/tasks/reports/data.go

```diff
@@ -194,11 +202,19 @@ func getVulnerabilityData(ctx context.Context, params sdkUtils.ReportParams) (*I
 		NodeWiseData:   nodeWiseData,
 	}
 
+	log.Info().Msgf("total vulnerability records in NodeWiseData is %d", data.NodeWiseData.RecordCount)
+
+	if data.NodeWiseData.RecordCount > sdkUtils.ReportRecordsMax {
+		return &data, ErrorMaxRecords
+	}
+
```

Comment:
We should bail off earlier, here we will store more than `sdkUtils.ReportRecordsMax` in memory.
We need to break out of the previous `for` loop.
Shall we also put a limit on the number of scans?